File: exercises/beecrowd/1045/1045.go
// https://judge.beecrowd.com.br/judge/pt/problems/view/1045
package main

import (
	"fmt"
	"math"
	"sort"
)

func sortDescending[T int | float64](array []T) []T {
	sort.Slice(array, func(i, j int) bool {
		return array[i] > array[j]
	})

	return array
}

func main() {
	var a, b, c float64

	fmt.Scanf("%f %f %f", &a, &b, &c)

	ordenedValues := sortDescending([]float64{a, b, c})

	a = ordenedValues[0]
	b = ordenedValues[1]
	c = ordenedValues[2]

	if a >= b+c {
		fmt.Println("NAO FORMA TRIANGULO")
		return
	}

	if math.Pow(a, 2) == math.Pow(b, 2)+math.Pow(c, 2) {
		fmt.Println("TRIANGULO RETANGULO")
	}

	if math.Pow(a, 2) > math.Pow(b, 2)+math.Pow(c, 2) {
		fmt.Println("TRIANGULO OBTUSANGULO")
	}

	if math.Pow(a, 2) < math.Pow(b, 2)+math.Pow(c, 2) {
		fmt.Println("TRIANGULO ACUTANGULO")
	}

	if a == b && b == c && c == a {
		fmt.Println("TRIANGULO EQUILATERO")
	} else if a == b || b == c || c == a {
		fmt.Println("TRIANGULO ISOSCELES")
	}
}

File: exercises/beecrowd/1045/1045.ts
// https://judge.beecrowd.com/pt/problems/view/1045
import * as fs from "fs";

function sortDescending<T extends number>(array: T[]): T[] {
    return array.sort((a, b) => b - a);
}

function main(): void {
	const input = fs.readFileSync("/dev/stdin", "utf8");
	const lines = input.split(/\r?\n/);

	const line = String(lines.shift()).split(' ');

	let a = Number(line[0])
	let b = Number(line[1])
	let c = Number(line[2])

	const ordenedValues = sortDescending([a, b, c])

	a = ordenedValues[0]
	b = ordenedValues[1]
	c = ordenedValues[2]

	if (a >= b+c) {
		console.log("NAO FORMA TRIANGULO")
		return
	}

	if (Math.pow(a, 2) == Math.pow(b, 2)+Math.pow(c, 2)) {
		console.log("TRIANGULO RETANGULO")
	}

	if (Math.pow(a, 2) > Math.pow(b, 2)+Math.pow(c, 2)) {
		console.log("TRIANGULO OBTUSANGULO")
	}

	if (Math.pow(a, 2) < Math.pow(b, 2)+Math.pow(c, 2)) {
		console.log("TRIANGULO ACUTANGULO")
	}

	if (a == b && b == c && c == a) {
		console.log("TRIANGULO EQUILATERO")
	} else if (a == b || b == c || c == a) {
		console.log("TRIANGULO ISOSCELES")
	}
}

main();
